types/block.go

```diff
@@ -386,30 +407,6 @@ func (commit *Commit) ValidateBasic() error {
 	if len(commit.Precommits) == 0 {
 		return errors.New("No precommits in commit")
 	}
-	height, round := commit.Height(), commit.Round()
-
-	// Validate the precommits.
-	for _, precommit := range commit.Precommits {
-		// It's OK for precommits to be missing.
-		if precommit == nil {
-			continue
-		}
-		// Ensure that all votes are precommits.
-		if precommit.Type != VoteTypePrecommit {
-			return fmt.Errorf("Invalid commit vote. Expected precommit, got %v",
-				precommit.Type)
-		}
-		// Ensure that all heights are the same.
-		if precommit.Height != height {
```

Comment:
shouldn't we still check this once for `commit`?

Reply:
Here the check is only working with data stored in the `Commit`, so it has no height/round to check against. The previous code was just checking that all contained precommits had the same values, but now they automatically do implicitly.